pkg/controller/cassandracluster/pod.go

```diff
@@ -111,6 +127,24 @@ func (rcc *ReconcileCassandraCluster) GetLastPod(namespace string, selector map[
 	return GetLastOrFirstPod(podsList, last)
 }
 
+// GetFirstPod returns the first pod satisfying the selector, being in the namespace and being ready
+func (rcc *ReconcileCassandraCluster) GetFirstPodReady(namespace string, selector map[string]string) (*v1.Pod, error) {
+	podsList, err := rcc.ListPods(namespace, selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cassandra's pods: %v", err)
+	}
+	return GetLastOrFirstPodReady(podsList.Items, first)
+}
+
+// GetLastPod returns the last pod satisfying the selector and being in the namespace
+func (rcc *ReconcileCassandraCluster) GetLastPodReady(namespace string, selector map[string]string) (*v1.Pod, error) {
+	podsList, err := rcc.ListPods(namespace, selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cassandra's pods: %v", err)
+	}
+	return GetLastOrFirstPodReady(podsList.Items, last)
+}
```

Comment:
why 2 different functions ? we can merge those two functions.

Reply:
It's to simplify the use, and don't have to be worry about, if we use 1 function, in which case it's First or Last. 
Same logic than the function `GetLastPod` and `GetFirstPod`.